src/app/pages/checkout/checkout-page.guard.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { Observable, debounceTime, map, of, switchMap, withLatestFrom } from 'rxjs';
+
+import { CheckoutFacade } from 'ish-core/facades/checkout.facade';
+import { whenFalsy } from 'ish-core/utils/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CheckoutPageGuard implements CanActivate {
+  constructor(private checkoutFacade: CheckoutFacade, private router: Router) {}
+
+  canActivate(): Observable<boolean | UrlTree> {
+    return this.checkoutFacade.basket$.pipe(
+      debounceTime(500),
```

Comment:
Why is the debounceTime needed? I think all possible scenearios, that a basket is loaded or is currently loading, are checked in the following switchMap. And this rxjs operator leads to the problem, that the checkout route is not activated for the first 500ms, although a basket could be available.

Reply:
No debounceTime needed anymore. Guard now handles the request with a race operator